src/btree/bt_read.c

```diff
@@ -125,17 +128,257 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
             upd = NULL;
             break;
         }
+        birthmarkp->instantiated = true;
     }
 
     /* We do not need the birthmark information in the lookaside structure anymore. */
-    for (i = 0; i < page_las->birthmarks_cnt; i++) {
-        birthmarkp = page_las->birthmarks + i;
+    for (i = 0, birthmarkp = page_las->birthmarks; i < page_las->birthmarks_cnt; i++, birthmarkp++)
         __wt_buf_free(session, &birthmarkp->key);
-    }
+
     page_las->birthmarks_cnt = 0;
     __wt_free(session, page_las->birthmarks);
 
 err:
+    for (i = 0, birthmarkp = page_las->birthmarks; i < page_las->birthmarks_cnt; i++, birthmarkp++)
+        birthmarkp->instantiated = false;
+
+    WT_TRET(__wt_btcur_close(&cbt, true));
+    __wt_free(session, upd);
+
+    return (ret);
+}
+
+/*
+ * __las_page_instantiate --
+ *     Instantiate lookaside update records that are not on disk image in a recently read page.
+ */
+static int
+__las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+    struct las_page_prepared_updates {
+        WT_ITEM key;
+        wt_timestamp_t timestamp;
+        uint64_t txnid;
+    } * las_preparep;
+    WT_BIRTHMARK_DETAILS *birthmarkp;
+    WT_CACHE *cache;
+    WT_CURSOR *cursor;
+    WT_CURSOR_BTREE cbt;
+    WT_DECL_ITEM(las_prepares);
+    WT_DECL_RET;
+    WT_ITEM las_key, las_value, next_las_key;
+    WT_PAGE *page;
+    WT_UPDATE *upd;
+    wt_timestamp_t durable_timestamp, las_timestamp;
+    size_t incr;
+    uint64_t birthmark_cnt, las_txnid, recno;
+    uint32_t i, las_btree_id, las_prepare_cnt, session_flags;
+    uint8_t prepare_state, upd_type;
+    const uint8_t *p;
+    int cmp, exact, instantiated_cnt;
+    bool birthmark_record, first_scan, locked;
+
+    birthmarkp = NULL;
+    cache = S2C(session)->cache;
+    cursor = NULL;
+    WT_CLEAR(las_key);
+    page = ref->page;
+    upd = NULL;
+    birthmark_cnt = 0;
+    recno = WT_RECNO_OOB;
+    las_btree_id = S2BT(session)->id;
+    las_prepare_cnt = 0;
+    session_flags = 0; /* [-Werror=maybe-uninitialized] */
+    instantiated_cnt = 0;
+    birthmark_record = false;
+    locked = false;
+
+    /*
+     * Check whether the disk image contains all the newest versions of the page or not?
```

Comment:
Comments should be statements. Please end it with a period and remove `or not?`.

Reply:
Fixed.